app/forms/mappings/Formatters.scala

```diff
@@ -222,4 +222,35 @@ trait Formatters extends Transforms {
 
     }
 
+  private[mappings] def mandatoryGIINFormatter(requiredKey: String,
+                                               lengthKey: String,
+                                               invalidKey: String,
+                                               formatKey: String,
+                                               regex: String,
+                                               formatRegex: String,
+                                               invalidCharKey: String,
+                                               validCharRegex: String
+  ): Formatter[String] =
+    new Formatter[String] {
+
+      override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], String] = {
+        val giin      = data.get(key).map(validGIINFormat)
+        val maxLength = 19
+
+        giin match {
+          case None | Some("")                               => Left(Seq(FormError(key, requiredKey)))
+          case Some(value) if value.length != maxLength      => Left(Seq(FormError(key, lengthKey)))
```

Comment:
Minor point but i don't think this variable should be called `maxLength` if we want our string to be of equal length to it

Reply:
Changed to setLength